File: client/src/types/domain/community.ts
export interface CommunityMember {
  id: number;
  userId: number;
  displayName: string;
  avatar?: string;
  role: "member" | "admin";
  joinedAt: string;
}

export interface Community {
  id: number;
  name: string;
  description: string;
  category?: string;
  batch?: string;
  department?: string;
  coverImage?: string;
  memberCount: number;
  isMember: boolean;
  createdBy: number;
  createdAt: string;
  updatedAt: string;
}

export interface CommunityMessage {
  id: number;
  communityId: number;
  userId: number;
  displayName: string;
  avatar?: string;
  content: string;
  createdAt: string;
}

export interface CreateCommunityRequest {
  name: string;
  description: string;
  category?: string;
  batch?: string;
  department?: string;
  isPublic?: boolean;
}

export interface PostCommunityMessageRequest {
  body: string;
}

File: server/alumniweb/src/main/java/com/alumniweb/alumniweb/model/Community.java
package com.alumniweb.alumniweb.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.SQLRestriction;
import org.hibernate.annotations.UpdateTimestamp;
import org.hibernate.annotations.SourceType;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
@Entity @Table(name = "community")
@SQLRestriction("deleted = false")
public class Community {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 200)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(length = 20)
    private String batch;

    @Column(length = 100)
    private String department;

    @Column(name = "created_by", nullable = false)
    private Long createdBy;

    @Column(name = "is_public")
    @Builder.Default
    private Boolean isPublic = true;

    @Column(name = "member_count")
    @Builder.Default
    private Integer memberCount = 0;

    @Transient
    @Builder.Default
    private Boolean isMember = false;

    @Builder.Default
    @Column(nullable = false)
    private boolean deleted = false;

    private LocalDateTime deletedAt;

    @OneToMany(mappedBy = "community", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private List<CommunityMember> members = new ArrayList<>();

    @CreationTimestamp(source = SourceType.DB)
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp(source = SourceType.DB)
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;
}
